File: examples/datasets/groundstations_custom.py
# /// script
# dependencies = ["brahe"]
# ///
"""
Creating custom groundstation data.

Demonstrates:
- Creating custom groundstation locations
- Adding properties to locations
- Combining custom with commercial networks
"""

import brahe as bh

if __name__ == "__main__":
    print("Custom Groundstation Data")
    print("=" * 60)

    # Create custom groundstation
    custom_station = (
        bh.PointLocation(
            lon=-122.4,  # degrees
            lat=37.8,  # degrees
            alt=100.0,  # meters
        )
        .add_property("provider", "Custom")
        .add_property("frequency_bands", ["S", "X", "Ka"])
        .add_property("name", "San Francisco Custom")
    )

    print("\nCustom station created:")
    print(f"  Location: {custom_station.lat:.2f}°N, {custom_station.lon:.2f}°E")
    print(f"  Altitude: {custom_station.alt:.1f} m")
    print(f"  Name: {custom_station.get_name()}")
    print(f"  Bands: {', '.join(custom_station.properties['frequency_bands'])}")

    # Combine with commercial network
    ksat_stations = bh.datasets.groundstations.load("ksat")
    all_stations = [custom_station] + ksat_stations

    print(f"\nTotal network size: {len(all_stations)} stations")
    print(f"  Commercial (KSAT): {len(ksat_stations)}")
    print("  Custom: 1")

    print("\n" + "=" * 60)
    print("Custom groundstation integration complete!")

File: examples/datasets/groundstations_custom.rs
//! Creating custom groundstation data.
//!
//! Demonstrates:
//! - Creating custom groundstation locations
//! - Adding properties to locations
//! - Combining custom with commercial networks

use brahe::datasets::groundstations;
use brahe::access::{PointLocation, AccessibleLocation};
use brahe::utils::identifiable::Identifiable;
use serde_json::json;

fn main() {
    println!("Custom Groundstation Data");
    println!("{}", "=".repeat(60));

    // Create custom groundstation
    let custom_station = PointLocation::new(
        -122.4,  // lon (degrees)
        37.8,    // lat (degrees)
        100.0    // alt (meters)
    )
    .with_name("San Francisco Custom")
    .add_property("provider", json!("Custom"))
    .add_property("frequency_bands", json!(["S", "X", "Ka"]));

    println!("\nCustom station created:");
    println!("  Location: {:.2}°N, {:.2}°E", custom_station.lat(), custom_station.lon());
    println!("  Altitude: {:.1} m", custom_station.alt());
    println!("  Name: {}", custom_station.get_name().unwrap_or("Unnamed"));
    if let Some(bands) = custom_station.properties().get("frequency_bands") {
        println!("  Bands: {}", bands);
    }

    // Combine with commercial network
    let ksat_stations = groundstations::load_groundstations("ksat").unwrap();
    let mut all_stations = vec![custom_station];
    all_stations.extend(ksat_stations.clone());

    println!("\nTotal network size: {} stations", all_stations.len());
    println!("  Commercial (KSAT): {}", ksat_stations.len());
    println!("  Custom: 1");

    println!("\n{}", "=".repeat(60));
    println!("Custom groundstation integration complete!");
}
